test/Microsoft.AspNetCore.Razor.Language.Test/VirtualRazorProjectFileSystemTest.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    public class VirtualRazorProjectFileSystemTest
```

Comment:
Going through these tests a lot of them are testing logic controlled by `DirectoryNode`, aka more integration tests than unit tests. I'm concerned that we're missing unit test coverage / testability of `VirtualRazorProjectFileSystem` inner workings.

Reply:
Sure